network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/ThreeWindingsTransformerAdderImpl.java

```diff
@@ -286,19 +265,30 @@ public ThreeWindingsTransformerAdder setRatedU0(double ratedU0) {
     public ThreeWindingsTransformer add() {
         String id = checkAndGetUniqueId();
 
-        if (leg1 == null) {
+        // Leg 1
+        if (legAdder1 == null) {
             throw new ValidationException(this, "Leg1 is not set");
         }
-        if (leg2 == null) {
+        VoltageLevel voltageLevel1 = legAdder1.checkVoltageLevel();
+        legAdder1.checkNodeBus();
+        LegAttributes leg1 = legAdder1.toLegAttributes();
```

Comment:
I think it would be better to have the two checks inside `toLegAttributes()`. This will prevent omitting them.